XamarinCommunityToolkit/Views/SegmentedView/Android/FormsSegments.android.cs

```diff
@@ -0,0 +1,294 @@
+		}
+
+		void Build()
+		{
+			strokeWidth = (int)context.ToPixels(defaultStrokeWidth);
+
+			SetBackgroundColor(backgroundColor);
+
+			Orientation = Orientation.Horizontal;
+			LayoutParameters = new LayoutParams(LayoutParams.MatchParent, LayoutParams.WrapContent);
+			CheckedChange += OnCheckChanged;
+		}
+
+		public void InitializeSegments()
+		{
+			if (ChildCount > 0)
+				RemoveAllViews();
+
+			for (var i = 0; i < Children.Count; i++)
+			{
+				var rb = InsertSegment(Children[i].ToString(), i);
+				if (i == 0)
+					rb.Checked = true;
+			}
+		}
+
+		RadioButton InsertSegment(string title, int index)
+		{
+			var isLeft = index == 0;
+			var position = isLeft ? Position.Left : index == Children.Count - 1 ? Position.Right : Position.Middle;
+			var rb = GetRadioButton(title, position);
+			AddView(rb);
+			return rb;
+		}
+
+		AContentRes.ColorStateList TextColorSelector => new AContentRes.ColorStateList(
+			new int[][] // states
+			{
+				new int[] { ARes.Attribute.StateChecked },
+				new int[] { -ARes.Attribute.StateChecked }
+			},
+			new int[] // colors
+			{
+				AColor.White, TintColor
+			});
+
+		AContentRes.ColorStateList SegmentColorSelector => new AContentRes.ColorStateList(
+			new int[][] // states
+			{
+				new int[] { ARes.Attribute.StateChecked },
+				new int[] { -ARes.Attribute.StateChecked }
+			},
+			new int[] // colors
+			{
+				TintColor, BackgroundColor
+			});
+
+		void OnCheckChanged(object sender, CheckedChangeEventArgs e)
+		{
+			CurrentSegment = FindViewById<RadioButton>(e.CheckedId);
+			SegmentSelected?.Invoke(this, new SelectedPositionChangedEventArgs(IndexOfChild(CurrentSegment)));
+		}
+
+		#region Drawable Resources
+
+		RadioButton GetRadioButton(string title, Position position)
+		{
+			BitmapDrawable icon;
+			var rb = new RadioButton(context);
+			rb.SetPadding(defaultButtonPadding, defaultButtonPadding, defaultButtonPadding, defaultButtonPadding);
+			rb.LayoutParameters = new RadioGroup.LayoutParams(0, LayoutParams.MatchParent, 1.0f);
+			rb.Gravity = GravityFlags.Center;
+			rb.SetButtonDrawable(null);
+
+			if(DisplayMode == SegmentMode.Image)
+			{
+				icon = GetImage(title);
+				rb.SetBackground(GetRadioButtonStateListDrawable(position, icon));
+				return rb;
+			}
+
+			rb.SetBackground(GetRadioButtonStateListDrawable(position));
+			rb.Text = title;
+			rb.TextAlignment = AViews.TextAlignment.Center;
+			rb.SetTextSize(ComplexUnitType.Sp, defaultTextSize);
+			rb.SetAllCaps(true);
+			rb.SetTypeface(null, TypefaceStyle.Bold);
+			rb.SetTextColor(TextColorSelector);
+			return rb;
+		}
+
+		BitmapDrawable GetImage(ImageSource filePath)
+		{
+			var tcs = new TaskCompletionSource<BitmapDrawable>();
+
+			Task.Run(async () =>
+			{
+				try
+				{
+					var result = await context.GetFormsBitmapAsync(filePath);
+					tcs.SetResult(new BitmapDrawable(Resources, result));
+				}
+				catch (Exception ex)
+				{
+					tcs.SetException(ex);
+				}
+			});
+
+			return tcs.Task.Result;
```

Comment:
Can this lead a `dead lock`?

Reply:
Since I am using `TaskCompletionSource` and don't want the `async` identifier, not sure how to get the result without doing `.Result`. Any ideas?

Reply:
I don't really get why you use `TaskCompletionSource` here. Normally it is used to make sync code async.
And you could have the same semantic doing:

```csharp
var task = Task.Run(async () => 
{
    var formsBitmap = await context.GetFormsBitmapAsync(filePath));
    return new BitmapDrawable(Resources, formsBitmap);
});

return task.GetAwaiter().GetResult();
```

Which is still bad as @pictos pointed it cause it could lead to deadlock or thread pool starvation.
I guess you should make your method async and adapt your code hierarchy accordingly: when it's async, it's async all the way :)

Reply:
Yeah, the @roubachof 's suggestion is valid, but since this code will run in a program that has an UIThread, this can be dangerous. So do we need this code finish to move to the next line or this code can run in parallel? If the logic need this peace to get done to do the next step, we need to rethink all the API and make it async friendly

Reply:
Nope won't be dangerous since the Synchronization context will be restored after the await statement. Also the iOS version has the same issue, just need to really go async till the root of the code :)

Reply:
Well, I was trying to avoid `async` chain due to a single `Task` and run it inside a synchronous method. But I see that it may cause an issue at some point. I am reviewing the API now. Great feedback, keep it coming.

Reply:
@hnabbasi if you need any help or want to discuss more, let us know.